rust/agents/relayer/src/relayer.rs

```diff
@@ -318,6 +319,18 @@ impl BaseAgent for Relayer {
             .instrument(info_span!("Relayer server"));
         tasks.push(server_task);
 
+        let txid_receivers = self
+            .message_syncs
+            .iter()
+            .filter_map(|(domain, sync)| {
+                let maybe_rx = sync.get_new_receive_tx_channel();
+                if maybe_rx.is_none() {
+                    warn!(?domain, "No txid receiver for chain");
```

Comment:
nit: wonder if we should just panic or something to make sure this never happens? as this is a startup invariant I assume we'll wanna keep. Don't feel strongly here

Reply:
unwrapping only makes sense if we want to enforce hook indexing. My assumption in this PR is that it's optional and we can alternate between this and cursors. If we stop relying on watermark cursor, can change this to a panic